pkg/server/auth_middleware.go

```diff
@@ -30,17 +29,15 @@ func (s *Server) AuthMiddleware() gin.HandlerFunc {
 
 		amr := AuthMiddlewareHeaderRequest{}
 		if err := c.BindHeader(&amr); err != nil {
-			respondWithError(c, http.StatusUnauthorized,
-				fmt.Sprintf("auth failed: %v", err))
+			respondWithError(c, NewError(http.StatusUnauthorized, err, "auth.middleware"))
```

Comment:
How about `c.Error(NewError(http.StatusUnauthorized, err, "auth.middleware")`?

Reply:
Personally, prefer keeping a func to do it. Because we can add common logic in this function.